### ISSUE TITLE
KNN, Decision Tree and NB

### ISSUE THREAD

**Issue by ArlonMoreira** [ArlonMoreira/DataManing#1]
#Biblioteca utilizada para obter arquivos csv
from pandas import read_csv
#Biblioteca utilizada para obter KNN
from sklearn.neighbors import KNeighborsClassifier
#Biblioteca de acuracia
from sklearn.metrics import accuracy_score
#Biblioteca para testar os dados
from sklearn.model_selection import train_test_split
#Biblioteca para calcular o recall
from sklearn.metrics import recall_score
#Biblioteca para calcular a precisão
from sklearn.metrics import precision_score
#Biblioteca para calcular matriz de decisão
from sklearn.metrics import confusion_matrix
#Bibliotece para árvore de decisão
from sklearn import tree
#Biblioteca Naive Bayes
from sklearn.naive_bayes import GaussianNB

#Gerando arquivo CSV
Car = read_csv('CarEvaluationDatabase.csv')
 
#Separa os atributos preditivos em X
Car.data = Car[['Precodecompra','Precodemanutencao','portas','pessoas','capacidadebagageiro','seguranca']]

#Separa atributo classe em Y
Car.target = Car['Class']

#Separa % dos dados para testar
X_train, X_test, y_train, y_test = train_test_split(Car.data, Car.target, test_size=0.33, random_state=42)

#======= KNN ======
#Cria uma instancia da classe KNeighborsClassifier() com os parâmetros
# setados em default, n_neighbors=5
knn = KNeighborsClassifier(9)

#Cria o modelo de classificacao
knn.fit(X_train,y_train)

#Testa o modelo nos dados de teste AQUI ELE VAI PREDIZER INFORMAR A CLASSE ....
predicted = knn.predict(X_test)

#Calcula a acuracia do modelo
acc = accuracy_score(y_test, predicted)    

#Calcula a precisão
precision = precision_score(y_test, predicted, average=None)

#Calcula o recall
recall = recall_score(y_test, predicted, average=None)  

#Matriz de decisão
matriz = confusion_matrix(y_test, predicted, labels=["acc", "good", "unacc", "vgood"])
#======= KNN ======

#====Arvore de decisão====

#Realizando treinamento
clf = tree.DecisionTreeClassifier()
clf = clf.fit(X_train,y_train)

#Testando o modelo
predictedtree = clf.predict(X_test)

acctree = accuracy_score(y_test, predictedtree) 

#Calcula a precisão
precisiontree = precision_score(y_test, predictedtree, average=None)

#Calcula o recall
recalltree = recall_score(y_test, predictedtree, average=None)  

#Matriz de decisão
matriztree = confusion_matrix(y_test, predictedtree, labels=["acc", "good", "unacc", "vgood"])

#======Naive Bayes======

#Treinamento
NB = GaussianNB()
NB.fit(X_train, y_train)

#Testando o modelo
predictedNB = NB.predict(X_test)

#Calculando a acurácia
accNB = accuracy_score(y_test, predictedNB)

#Precisão
precisionNB =  precision_score(y_test, predictedNB, average=None)

#Recall
recallNB = recall_score(y_test, predictedNB, average=None)  

#Matriz
matrizNB = confusion_matrix(y_test, predictedNB, labels=["acc", "good", "unacc", "vgood"])

print("Acurácia (KNN):")
print(acc)
print('Precisão (KNN):')
print(precision)
print('Recall (KNN):')
print(recall)
print('Matriz (KNN):')
print(matriz)
print("Acurácia (Decision Tree):")
print(acctree)
print('Precisão (Decision Tree):')
print(precisiontree)
print('Recall (Decision Tree):')
print(recalltree)
print('Matriz (Decision Tree):')
print(matriztree)
print("Acurácia (Naive Bayes):")
print(accNB)
print('Precisão (Naive Bayes):')
print(precisionNB)
print('Recall (Naive Bayes):')
print(recallNB)
print('Matriz (Naive Bayes):')
print(matrizNB)
